src/pytorch_lightning/trainer/trainer.py

```diff
@@ -567,66 +563,64 @@ def _init_debugging_flags(
+                self.limit_train_batches = num_batches
+                self.limit_val_batches = num_batches
+
+            self.limit_test_batches = num_batches
+            self.limit_predict_batches = num_batches
             self.fit_loop.max_steps = num_batches
             self.num_sanity_val_steps = 0
             self.fit_loop.max_epochs = 1
-            val_check_interval = 1.0
+            self.val_check_interval = 1.0
             self.check_val_every_n_epoch = 1
             self.loggers = [DummyLogger()] if self.loggers else []
-
             rank_zero_info(
                 f"Running in `fast_dev_run` mode: will run the requested loop using {num_batches} batch(es). "
                 "Logging and checkpointing is suppressed."
             )
-
-        self.limit_train_batches = _determine_batch_limits(limit_train_batches, "limit_train_batches")
-        self.limit_val_batches = _determine_batch_limits(limit_val_batches, "limit_val_batches")
-        self.limit_test_batches = _determine_batch_limits(limit_test_batches, "limit_test_batches")
-        self.limit_predict_batches = _determine_batch_limits(limit_predict_batches, "limit_predict_batches")
-        self.val_check_interval = _determine_batch_limits(val_check_interval, "val_check_interval")
-        self.overfit_batches = _determine_batch_limits(overfit_batches, "overfit_batches")
-        self._configure_overfit_batches(self.overfit_batches)
-
-    def _configure_overfit_batches(self, overfit_batches: Union[int, float]) -> None:
-        """Configure batch limits using `overfit_batches`."""
-        if overfit_batches > 0:
+        else:
+            if not overfit_batches_enabled:
+                self.limit_train_batches = _determine_batch_limits(limit_train_batches, "limit_train_batches")
+                self.limit_val_batches = _determine_batch_limits(limit_val_batches, "limit_val_batches")
+            self.limit_test_batches = _determine_batch_limits(limit_test_batches, "limit_test_batches")
+            self.limit_predict_batches = _determine_batch_limits(limit_predict_batches, "limit_predict_batches")
+            self.num_sanity_val_steps = float("inf") if num_sanity_val_steps == -1 else num_sanity_val_steps
+            self.val_check_interval = _determine_batch_limits(val_check_interval, "val_check_interval")
+
+        if overfit_batches_enabled:
             self.limit_train_batches = overfit_batches
             self.limit_val_batches = overfit_batches
 
-    def _setup_on_init(self, num_sanity_val_steps: int) -> None:
+    def _setup_on_init(self) -> None:
         self._log_device_info()
 
         self.should_stop = False
         self.state = TrainerState()
         self.num_training_batches = float("inf")
         self.train_dataloader = None
 
-        if num_sanity_val_steps == -1:
-            self.num_sanity_val_steps = float("inf")
-        else:
-            self.num_sanity_val_steps = num_sanity_val_steps
```

Comment:
I suppose this was an unrelated refactor because you want to classify sanity_val_steps as a debugging flag?

Reply:
I believe I made this change. I moved it because in one of the `_init_debugging_flags`'s branches we were already setting `self.num_sanity_val_steps = 0`, so it made sense that we managed the attribute entirely inside that method